### PR TITLE
Migrate setup configuration to setup.cfg

### PR DESCRIPTION
**Pull request by DragaDoncila**
- Moved configuration options into `setup.cfg`.
- For editable installs, kept `use_scm` in `setup.py`.
-  `setup.cfg` doesn't support a `file:` option for the `install_requires` keyword, and it doesn't look like that'll be coming anytime soon, based on the discussion in [this issue](https://github.com/pypa/setuptools/issues/1951). So for our purposes I've kept the requirements reading in setup.py

I built a plugin locally and it worked as expected. I tried running tests locally but running the created plugin's tests failed with 
```python
E           TypeError: Failed expected string as 'msg' parameter, got 'CalledProcessError' instead.
E           Perhaps you meant to use a mark?
```

I saw the same behaviour on master though so I think it must be something I'm doing wrong.

Addresses #26.

### CODE REVIEW

**Comment by tlambert03** [comment]
> setup.cfg doesn't support a file: option for the install_requires keyword, and it doesn't look like that'll be coming anytime soon, based on the discussion in this issue. So for our purposes I've kept the requirements reading in setup.py

let's move away from reading `requirements.txt` into the `setup.cfg`... and do it the other way around instead:

```txt
# requirements.txt

-e .
```

**Comment by DragaDoncila** [comment]
@tlambert03 thanks for the details - I know they're very newbie questions. If I've understood you correctly, then these latest two commits should do it.

File: {{cookiecutter.plugin_name}}/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    License :: OSI Approved :: GNU General Public License v3 (GPLv3)
+    {%- elif cookiecutter.license == "Apache Software License 2.0" -%}
+    License :: OSI Approved :: Apache Software License
+    {%- elif cookiecutter.license == "Mozilla Public License 2.0" -%}
+    License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)
+    {%- endif %}
+
+[options]
+packages = find:
+python_requires = >=3.6
```
Comment (tlambert03) [inline {{cookiecutter.plugin_name}}/setup.cfg:35]:
```suggestion
python_requires = >=3.7
```

why not :)

File: {{cookiecutter.plugin_name}}/setup.cfg [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+setup_requires = setuptools_scm
+# add your minimal requirements here
```
Comment (tlambert03) [inline {{cookiecutter.plugin_name}}/setup.cfg:37]:
```suggestion
# add your package requirements here
```

File: {{cookiecutter.plugin_name}}/setup.cfg [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+name = {{cookiecutter.plugin_name}}
+author = {{cookiecutter.full_name}}
+author_email = {{cookiecutter.email}}
+license = {{cookiecutter.license}}
+url = https://github.com/{{cookiecutter.github_username}}/{{cookiecutter.plugin_name}}
+description = {{cookiecutter.short_description}}
+long_description = file: README.md
+long_description_content_type = text/markdown
+classifiers =
+    Development Status :: 4 - Beta
```
Comment (tlambert03) [inline {{cookiecutter.plugin_name}}/setup.cfg:11]: let's change this now too:

```suggestion
    Development Status :: 2 - Pre-Alpha
```

**Review by tlambert03 — APPROVED** [review]

**Review by jni — APPROVED** [review]

**Pull request merged**